wundergraph_cli/src/print_schema/mod.rs

```diff
@@ -311,41 +312,51 @@ mod tests {
         std::thread::sleep(std::time::Duration::from_secs(1));
 
         let query = "{\"query\": \"{ Users { id  name  } } \"}";
-        let mut r = client
-            .post(&format!("http://{}/graphql", listen_url))
-            .body(query)
-            .header(
-                reqwest::header::CONTENT_TYPE,
-                reqwest::header::HeaderValue::from_static("application/json"),
-            )
-            .send()
-            .unwrap();
-        insta::assert_json_snapshot!(r.json::<serde_json::Value>().unwrap());
-
         let mutation = r#"{"query":"mutation CreateUser {\n  CreateUser(NewUser: {name: \"Max\"}) {\n    id\n    name\n  }\n}","variables":null,"operationName":"CreateUser"}"#;
+        let t1 = request_test(&client, &listen_url, query);
+        let t2 = request_test(&client, &listen_url, mutation);
+        let t3 = request_test(&client, &listen_url, query);
+
+        child.kill().unwrap();
+        child.wait().unwrap();
+
+        if !t1 || !t2 || !t3 {
+            panic!("round_trip failed")
+        }
+    }
+
+    fn request_test(
+        client: &reqwest::Client,
+        listen_url: &str,
+        body: &'static str,
+    ) -> bool {
         let mut r = client
             .post(&format!("http://{}/graphql", listen_url))
-            .body(mutation)
+            .body(body)
             .header(
                 reqwest::header::CONTENT_TYPE,
                 reqwest::header::HeaderValue::from_static("application/json"),
             )
             .send()
             .unwrap();
-        insta::assert_json_snapshot!(r.json::<serde_json::Value>().unwrap());
 
-        let mut r = client
-            .post(&format!("http://{}/graphql", listen_url))
-            .body(query)
-            .header(
-                reqwest::header::CONTENT_TYPE,
-                reqwest::header::HeaderValue::from_static("application/json"),
-            )
-            .send()
+        let builder =
+            thread::Builder::new().name("insta-assert-json-snapshot".into());
+        let handler = builder
+            .spawn(move || {
+                let js = r.json::<serde_json::Value>().unwrap();
+                insta::assert_json_snapshot!(js.clone())
+            })
             .unwrap();
-        insta::assert_json_snapshot!(r.json::<serde_json::Value>().unwrap());
 
-        child.kill().unwrap();
-        child.wait().unwrap();
+        match handler.join() {
```

Comment:
As this is test code it's totally fine to just do a `dbg!(handler.join())` here. Cargo does not print output from succeeding tests by default.

So something like `dbg!(handler.join()).is_ok()` should be fine.

Reply:
375c03aea376cb057d3261145ee1344ed076fba5 uses a closure for two `Err(..)`'s inside `request_test`. Do you still see demand of amendment?